lib/server/db/mysql/mysql.go

```diff
@@ -93,6 +95,18 @@ func (m *Mysql) PingContext(ctx context.Context) error {
 	return nil
 }
 
+// exists determines if the database has already been created
+func exists(sqlxDB db.FabricCADB, dbName string) (bool, error) {
+	log.Debugf("Checking if MySQL CA Database (%s) exists", dbName)
+	var exists bool
+	query := fmt.Sprintf("SELECT true as 'exists' FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = '%s'", dbName)
+	err := sqlxDB.Get("CheckIfDatabaseExists", &exists, query)
+	if err != nil && err != sql.ErrNoRows {
+		return false, errors.Wrapf(err, "Failed to check if MySQL database '%s' exists", dbName)
+	}
+	return exists, nil
+}
```

Comment:
Why is this needed for MySQL? The CREATE DATABASE statement supports the IF NOT EXISTS predicate for MySQL and it's already used in the createDatabase function below.

I thought the reported problem was for postgres where the IF NOT EXISTS predicate is not supported?

Reply:
I tested this. If the user has the `CREATE` permission on the database which grants `Database, Table and Column` permission, `IF NOT EXISTS` works. But if the user only has `Insert` permission they have  `Table and Column` permission and `IF NOT EXISTS` returns `Access denied for user 'btl5037'@'%' to database 'test'`. With create they can modify the existing database. I imagine if you were locking down permissions you would provide `Insert` and not `Create` permission. But I could be wrong, thought?